Analysis/Tasks/PWGHF/HFTreeCreatorD0ToKPi.cxx

```diff
@@ -0,0 +1,263 @@
+      candidate.index0_as<aod::BigTracksPID>().tpcNSigmaKa(),                                                           \
+      candidate.index0_as<aod::BigTracksPID>().tofNSigmaPi(),                                                           \
+      candidate.index0_as<aod::BigTracksPID>().tofNSigmaKa(),                                                           \
+      candidate.index1_as<aod::BigTracksPID>().tpcNSigmaPi(),                                                           \
+      candidate.index1_as<aod::BigTracksPID>().tpcNSigmaKa(),                                                           \
+      candidate.index1_as<aod::BigTracksPID>().tofNSigmaPi(),                                                           \
+      candidate.index1_as<aod::BigTracksPID>().tofNSigmaKa(),                                                           \
+      1 << CandFlag,                                                                                                    \
+      FunctionInvMass(candidate),                                                                                       \
+      candidate.impactParameterProduct(),                                                                               \
+      FunctionCosThetaStar(candidate),                                                                                  \
+      candidate.pt(),                                                                                                   \
+      candidate.p(),                                                                                                    \
+      candidate.cpa(),                                                                                                  \
+      candidate.cpaXY(),                                                                                                \
+      FunctionCt(candidate),                                                                                            \
+      candidate.eta(),                                                                                                  \
+      candidate.phi(),                                                                                                  \
+      FunctionY(candidate),                                                                                             \
+      FunctionE(candidate),                                                                                             \
+      candidate.flagMCMatchRec());                                                                                      \
+  }
+
+      FILL_TREE(0, isSelD0, InvMassD0, CosThetaStarD0, CtD0, YD0, ED0);
+      FILL_TREE(1, isSelD0bar, InvMassD0bar, CosThetaStarD0bar, CtD0, YD0, ED0);
+    }
+
+    // Filling particle properties
+    rowCandidateFullParticles.reserve(particles.size());
+    for (auto& particle : particles) {
+      if (particle.flagMCMatchGen() == DecayType::D0ToPiK) {
```

Comment:
@ginnocen this is now fixed thanks to @vkucera (same for Lc)

Reply:
Why do you reject D0 bar?

Reply:
i'm afraid it's the same 
```
// mapping of decay types
enum DecayType { D0ToPiK = 0,
                 JpsiToEE,
                 N2ProngDecays }; //always keep N2ProngDecays at the end
```

Reply:
This is going to be fixed in https://github.com/AliceO2Group/AliceO2/pull/5330
Don't get distracted by my bugs. :-)
Your code would not work for anti-Lambda_c anyway.

Reply:
I fixed it manually Vit